lib/PoolCluster.js

```diff
@@ -124,17 +128,35 @@ PoolCluster.prototype._getNode = function(id) {
 };
 
 PoolCluster.prototype._increaseErrorCount = function(node) {
+  var self = this;
+
   if (++node.errorCount >= this._removeNodeErrorCount) {
     var index = this._serviceableNodeIds.indexOf(node.id);
     if (index !== -1) {
       this._serviceableNodeIds.splice(index, 1);
-      delete this._nodes[node.id];
-
       this._clearFindCaches();
 
-      node.pool.end();
+      if ((this._restoreNodeTimeout > 0) && !node.restoreTimeoutID) {
+        node.online = false;
+        this.emit('offline', node.id);
+
+        node.restoreTimeoutID = setTimeout(function() {
```

Comment:
Hi-res time is good idea.
Then, we should create another queue `_offlineNodeIds` (like `_serviceableNodeIds`). Every time we call `_findNodeIds`, it will check first record of `_offlineNodeIds` (id+hrtime) and restore offline node if it is needed. Then restored node should be moved from `_offlineNodeIds` to `_serviceableNodeIds`.

Do you approve this idea?